source/agora/common/Deserializer.d

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+
+    Function definition and helper related to Deserialization
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Deserializer;
+
+import agora.common.Data;
+import agora.common.Block;
+import agora.common.crypto.Key;
+
+/// Type of delegate deserializeDg
+public alias DeserializeDg = ubyte[] delegate(size_t size) nothrow @safe;
+
+/*******************************************************************************
+
+    Deserialize a struct and return it as a Block.
+
+    Params:
+        T = Type of struct to deserialize
+        record = Instance of `T` to deserialize
+        dg  = Deserialization delegate, when this struct is a nested struct
```

Comment:
`source` is undocumented

Reply:
record -> source = uByte[] source to be deserialized